config/v1/types_ingress.go

```diff
@@ -152,8 +153,75 @@ type AWSIngressSpec struct {
 	// +kubebuilder:validation:Enum:=NLB;Classic
 	// +kubebuilder:validation:Required
 	Type AWSLBType `json:"type,omitempty"`
+
+	// networkLoadBalancerParameters holds configuration parameters for an AWS
+	// Network Load Balancer. This field is optional and permitted only if the type field is set to NLB.
+	//
+	// +openshift:enable:FeatureGate=SetEIPForNLBIngressController
+	// +optional
+	NetworkLoadBalancerParameters *AWSNetworkLoadBalancerParameters `json:"networkLoadBalancer,omitempty"`
 }
 
+// AWSNetworkLoadBalancerParameters holds configuration parameters for an
+// AWS Network Load Balancer.
+type AWSNetworkLoadBalancerParameters struct {
+	// eipAllocations assign Elastic IP addresses to the Network Load Balancer.
+	// The number of Allocation IDs must match the number of subnets that are used for the load balancer.
+	// See https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html for general
+	// information about configuration, characteristics, and limitations of Elastic IP addresses.
+	// See the documentation for the IngressController spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.networkLoadBalancer.eipAllocations
+	// API field for limitations on the use of Elastic IP addresses in the context of IngressControllers.
+	// This field only specifies EIPs for the default IngressController. It is not a default value
+	// for all IngressControllers.
+	// If the cluster config and the default IngressController both specify eipAllocations and if we manually create the default
+	// IngressController before the polling loop of Ingress Operator to check existence of default IngressController to spin an
+	// Ingress Controller from with the values from config object then default IC from manual creation will be created with the eipAllocations
+	// mentioned in it.
+	//
+	// +optional
+	// +listType=atomic
+	// Each entry in the eipAllocations list must be unique
+	// +kubebuilder:validation:XValidation:rule=`self.all(x, self.exists_one(y, x == y))`,message="eipAllocations cannot contain duplicates"
+	// A maximum of 10 IP allocations are permitted
```

Comment:
Please add "." at the end of sentences.  Make sure you check the generated CRD.

Reply:
ok done.